pr2eus/robot-interface.l

```diff
@@ -190,19 +190,45 @@
                         (t joint-states-topic))
                      msg))
      msg))
+
+  (:angle-vector-duration 
+   (end &optional (scale 5) (min-time 1.0))
+   (let ((joint-list (send robot :joint-list))
+	 (start (send self :state :potentio-vector))
+	 )
+     (let ((diff (coerce (v- end start) cons)))
+       (let ((time-list (mapcar #'(lambda (d j)
+				    (* scale (/ (deg2rad (abs d)) (send j :max-joint-velocity))))
+				diff joint-list)))
+	 (let ((max-time (apply #'max time-list)))
+	   (max max-time min-time))))))
+
   (:angle-vector
-   (av &optional (tm 3000) (ctype controller-type) (start-time 0))
-   ;; check max-joint-velocity
-   (let ((idx 0)
-         (diff-av (v- av (or (send self :state :potentio-vector) (send robot :angle-vector)))))
-     (dolist (jt (send robot :joint-list))
-       (cond ((derivedp jt linear-joint) ;; msec = mm / [m/sec]
-              (setq tm (max tm (abs (/ (elt diff-av idx) (send jt :max-joint-velocity))))))
-             ((derivedp jt rotational-joint) ;; msec = deg2rad(deg) / [rad/s] / 0.001
-              (setq tm (max tm (abs (/ (deg2rad (elt diff-av idx))
-                                       (send jt :max-joint-velocity) 0.001))))))
-       (incf idx)
-       ))
+   ;;tm (time to goal in msec)
```

Comment:
comment shouldd be "...", see 
for example https://github.com/euslisp/jskeus/pull/138/files#diff-84175dc49eace6b214a353e6d2b5fb34R923 goes to http://euslisp.github.io/jskeus/jmanual-node170.html#52679